extends Control

func _ready():
	var data = Global.get_all_player_data()
	if not Global.is_logged_in:
		return
	#if data:
		#print(data)
	if data.image != null:
		var base64_string = data.image
		base64_string = clean_base64(base64_string)  # Nettoie la chaÃ®ne base64
		var image_texture = load_image_from_base64(base64_string)
		#print("ğŸ” Base64 reÃ§u (50 premiers caractÃ¨res) :", image_texture.substr(0, 50))
  
		$ProfileImage.texture = image_texture
		$ProfileImage.visible = true
		$ProfileImage.queue_redraw()

	$ProfileImage/Pseudo.text = data.pseudo

	$CreateBtn.connect("pressed", _on_start_pressed)
	$JoinBtn.connect("pressed", _on_join_pressed)
	$DisconnectBtn.connect("pressed", _on_disconnect_pressed)
	
	Global.pull_list_dominos()
	

func _on_start_pressed():
	get_tree().change_scene_to_file("res://Scenes/create_game.tscn")

func _on_join_pressed():
	get_tree().change_scene_to_file("res://Scenes/listeSessions.tscn")
	
func _on_disconnect_pressed():
	Global.reset_player()
	get_tree().change_scene_to_file("res://Scenes/principal.tscn")
	
func load_image_from_base64(base64_str: String) -> ImageTexture:
	var image = Image.new()
	var img_data = Marshalls.base64_to_raw(base64_str)

	var err = ERR_PARSE_ERROR
	if base64_str.begins_with("/9j/"):  # JPEG
		err = image.load_jpg_from_buffer(img_data)
	else:  # PNG
		err = image.load_png_from_buffer(img_data)

	if err != OK:
		print("âŒ Erreur lors du chargement de l'image :", err)
		return null

	# Appel statique correct pour Godot 4.x
	var texture = ImageTexture.create_from_image(image)
	return texture



func clean_base64(base64_str: String) -> String:
	# Supprime le "b'" au dÃ©but et "'" Ã  la fin si prÃ©sent
	base64_str = base64_str.strip_edges()  # Supprime les espaces et apostrophes

	# VÃ©rifie s'il commence par "b'" et enlÃ¨ve ce prÃ©fixe
	if base64_str.begins_with("b'") or base64_str.begins_with('b"'):
		base64_str = base64_str.substr(2, base64_str.length() - 3)  # Supprime les b' et '

	# VÃ©rifie s'il commence par "data:image"
	if base64_str.begins_with("data:image"):
		base64_str = base64_str.split(",")[1]  # EnlÃ¨ve "data:image/png;base64,"

	return base64_str
#func load_image_from_base64(base64_str: String) -> ImageTexture:
	#var image = Image.new()
	#var img_data = Marshalls.base64_to_raw(base64_str)  # Convertir le Base64 en donnÃ©es binaires
	#var err = image.load_png_from_buffer(img_data)  # Charger les donnÃ©es binaires en image
#
	#if err != OK or not image.is_valid():
		#print("âŒ Erreur de dÃ©codage de lâ€™image :", err)
		#return null
#
	#var texture = ImageTexture.new()
	#texture.create_from_image(image)
	#return texture
